docs/api.md

```diff
@@ -196,18 +221,57 @@ type NodePoolSpec struct {
     MachineConfig MachineConfigSpec
 
     Template MachineConfigTemplate
+
+    Platform NodePoolPlatform
+}
+
+// NodePoolPlatform is the platform-specific configuration for a node
+// pool. Only one of the platforms should be set.
+type NodePoolPlatform struct {
+	// AWS is the configuration used when installing on AWS.
+	AWS *AWSNodePoolPlatform `json:"aws,omitempty"`
+}
+
+// AWSNodePoolPlatform stores the configuration for a node pool
+// installed on AWS.
+type AWSNodePoolPlatform struct {
+	// InstanceType defines the ec2 instance type.
+	// eg. m4-large
+	InstanceType    string                `json:"instanceType"`
+	InstanceProfile string                `json:"instanceProfile,omitempty"`
+	Subnet          *AWSResourceReference `json:"subnet,omitempty"`
+        SecurityGroups  []string              `json:securityGroups,omitempty"`
```

Comment:
wrong indentation

Reply:
is this a good spot for an AMI override?

we will need this for billable AMI from AWS marketplace

https://github.com/openshift/installer/pull/2002/files

Reply:
Not sure about the concept of an "override" — maybe a clarifying question would be: what is the source of truth for the AMI? Should it be explicitly declared on spec and filled in automatically with a default if empty? etc.

Reply:
@ironcladlou that makes sense to me. If empty, we set a default.